internal/cmd/skupper/debug/nonkube/debug.go

```diff
@@ -21,13 +45,355 @@ func NewCmdDebug() *CmdDebug {
+	lopts := fs.GetOptions{ResourcesPath: lpath, RuntimeFirst: true, LogWarning: false}
+	links, err = cmd.linkHandler.List(lopts)
+	if err == nil && links != nil && len(links) != 0 {
+		for _, link := range links {
+			err := utils.WriteObject(link, path+link.Name+link.Spec.Endpoints[0].Host, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	//runtime/resources
+	path = rpath + "resources/"
+	opts = fs.GetOptions{RuntimeFirst: true, LogWarning: false, RemoveKey: true}
+	sites, err = cmd.siteHandler.List(opts)
+	if err == nil && sites != nil && len(sites) != 0 {
+		for _, site := range sites {
+			err := utils.WriteObject(site, path+"Site-"+site.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	routerAccesses, err = cmd.routerAccessHandler.List(opts)
+	if err == nil && routerAccesses != nil && len(routerAccesses) != 0 {
+		for _, routerAccess := range routerAccesses {
+			err = utils.WriteObject(routerAccess, path+"RouterAccess-"+routerAccess.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	listeners, err = cmd.listenerHandler.List(opts)
+	if err == nil && listeners != nil && len(listeners) != 0 {
+		for _, listener := range listeners {
+			err := utils.WriteObject(listener, path+"Listener-"+listener.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	connectors, err = cmd.connectorHandler.List(opts)
+	if err == nil && connectors != nil && len(connectors) != 0 {
+		for _, connector := range connectors {
+			err := utils.WriteObject(connector, path+"Connector-"+connector.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	certificates, err := cmd.certificateHandler.List()
+	if err == nil && certificates != nil && len(certificates) != 0 {
+		for _, certificate := range certificates {
+			err := utils.WriteObject(certificate, path+"Certificate-"+certificate.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	secrets, err = cmd.secretHandler.List(opts)
+	if err == nil && secrets != nil && len(secrets) != 0 {
+		for _, secret := range secrets {
+			err := utils.WriteObject(secret, path+"Secret-"+secret.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	configMaps, err := cmd.configMapHandler.List()
+	if err == nil && configMaps != nil && len(configMaps) != 0 {
+		for _, configMap := range configMaps {
+			err := utils.WriteObject(configMap, path+"ConfigMap-"+configMap.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	//runtime/router
+	path = rpath + "router/"
+	skrPath := api.GetDefaultOutputPath(cmd.namespace) + "/" + string(api.RouterConfigPath+"/skrouterd.json")
+	//skrPath := pathProvider.GetRuntimeNamespace()+ api.
+	skrouterd, err := os.ReadFile(skrPath)
+	if err == nil && skrouterd != nil {
+		utils.WriteTar(path+"skrouterd.json", skrouterd, time.Now(), tw)
+	}
+
+	//logs and skupper-router statistics
+	cli, err := internalclient.NewCompatClient(os.Getenv("CONTAINER_ENDPOINT"), "")
```

Comment:
Before creating the compat client, set the `SKUPPER_PLATFORM` environment variable, using the `platform` loaded using the `NamespacePlatformLoader`.

Reply:
done